docs/experimental-features/investigation-guide-actions.asciidoc

```diff
@@ -0,0 +1,80 @@
+[[interactive-investigation-guides]]
+== Interactive investigation guides
+
+NOTE: This feature is available for {stack} versions 8.6.0 and newer.
+
+Detection rule investigation guides suggest steps for triaging, analyzing, and responding to potential security issues. For custom rules, you can create an interactive investigation guide that includes buttons for launching runtime queries in <<timelines-ui,Timeline>>, using alert data and hard-coded literal values. This allows you to launch detailed Timeline investigations directly from an alert using relevant data. 
+
+[role="screenshot"]
+image::images/ig-alert-flyout.png[Alert details flyout with interactive investigation guide,550]
+
+Each query button displays the number of documents found. Click the button to automatically create the query in Timeline based on the investigation guide's configuration.
+
+[role="screenshot"]
+image::images/ig-timeline.png[Timeline with query pre-loaded from investigation guide action]
+
+[discrete]
+[[add-ig-actions-rule]]
+=== Add investigation guide actions to a rule
+
+NOTE: You can only create interactive investigation guides with custom rules because Elastic prebuilt rules can't be edited. However, you can duplicate a prebuilt rule, then configure the investigation guide for the duplicated rule.
+
+You can configure an interactive investigation guide when you <<rules-ui-create,create a new rule>> or <<edit-rules-settings,edit an existing rule>>. When configuring the rule's settings (the *About rule* step for a new rule, or the *About* tab for an existing rule), expand the *Advanced settings*, then scroll down to the *Investigation guide* field.
+
+[role="screenshot"]
+image::images/ig-investigation-guide-editor.png[Investigation guide editor field]
+
+Add the following syntax to the *Investigation guide* field to configure a query. The syntax will appear as an action button in the investigation guide.
+
+[width="100%",options="header"]
+|===
+|Field |Description
+
+|`!{insight{ }}` |The container object holding all the configuration attributes for the button and the query.
+|`label` |Label text on the button.
+|`description` |Description text on the button.
+|`providers` a|A two-level nested array that defines the query you want to run in Timeline. Similar to the structure of queries in Timeline, items in the outer level are joined by an `OR` relationship, and items in the inner level are joined by an `AND` relationship.
+
+Each item in `providers` is defined by these attributes:
+
+* `field`: The name of a field to query.
+* `value`: The value to search for. Either a hard-coded literal value, or the name of the field on the alert whose value you want to use as a query parameter.
```

Comment:
We should mention here that any value that contains a \ character need to be double escaped, i.e. a windows file path would be "C:\\\\Windows\\\\System32". This is a bit gross without the form builder doing this for them, but won't be an issue next release. The reason for this is that JSON.stringify/JSON.parse is ultimately used twice, once on the entire markdown snippet, and once on the provider part of the snippet.

Reply:
ha the github markdown behaves similarly, that was supposed to be 4 slashes delineating each part of the path, so I think it ends up being 8. "C:\\\\\\\\Windows\\\\\\\\System32"

Reply:
Thanks @kqualters-elastic! Are there any other special characters that need to be escaped, either double `\\`or single `\`?

Reply:
Ya I believe all the other JSON special characters as well:
\b  Backspace (ascii code 08)
\f  Form feed (ascii code 0C)
\n  New line
\r  Carriage return
\t  Tab
\"  Double quote
\\  Backslash character

Reply:
@kqualters-elastic I added a [note](https://github.com/elastic/security-docs/pull/2821/commits/a95f1fdd39cc428b4fcf0019226996d8cff66b3f) to explain escape characters, based on our Slack discussion. It's intentionally a little more general, pointing to the error icon to help users troubleshoot syntax errors. Feel free to edit as needed, once you've confirmed how exactly the characters work.